### ISSUE TITLE
Accessibility: docs code examples/link styling is very confusing and has insufficient contrast

### ISSUE THREAD

**Issue by sm-kc** [reactjs/reactjs.org#6111]
In the documentation, the styling is really confusing.

Here's an example, taken from [the docs for `createContext`](https://react.dev/reference/react/createContext) in dark mode, although the issue is present in light mode as well:

<img width="790" alt="image" src="https://github.com/reactjs/react.dev/assets/92527868/102b0e74-da8b-4ae6-b924-4c48271571da">

Here you can see:
- a text link ("pass information deep down") which is blue and not underlined and has no background color
- variables in source code, which are a similar shade of blue on a darker blue background, with a bottom box border, and aren't clickable
- string values in source code, same as above but purple on purple
- a concept in text ("context object"), variable width text, same colorations as source variables, not clickable
- another concept ("default values") same as above but purple
- a text link which is a function name (`useContext()`), blue monospaced text on a gray background

This assortment of similar-but-different color combinations is making it hard for me to read. Everything with a box bottom border looks like a link and I keep trying to click them out of habit that they're a crossreference. The use of blue and purple for "context object" and "default values" also add to the problem because they look like unvisited/visited links. At the very least, could you please consider changing either your source code highlights or your link color to be something drastically other than blue?

Running an accessibility checker on the pages also shows that many of the styles fail WCAG accessibility standards for text contrast.

<img width="537" alt="image" src="https://github.com/reactjs/react.dev/assets/92527868/be5cbcf7-79e4-4713-9181-31fed16c1073">
